transport/http/src/main/java/org/kaazing/gateway/transport/http/PersistentConnectionPool.java

```diff
@@ -85,46 +113,92 @@ public void recycle(IoSession session, Integer keepAliveTimeout) {
      * @return a reusable IoSession for the address
      *         otherwise null
      */
-    public IoSession take(ResourceAddress address) {
-        Set<IoSession> sessions = connections.get(address);
-        if (sessions != null && !sessions.isEmpty()) {
-            IoSession session = sessions.iterator().next();
-            sessions.remove(session);
+    IoSession take(HttpResourceAddress serverAddress) {
+        IoSession transportSession = removeThreadAligned(serverAddress);
+        if (transportSession != null) {
+            // Got a cached persistent connection
 
-            if (logger.isDebugEnabled()) {
-                logger.debug(String.format("Reusing (removing from pool) http connect persistent connection %s", session));
+            // Remove session idle tracking for this session
+            transportSession.getConfig().setBothIdleTime(0);
+            IoFilterChain filterChain = transportSession.getFilterChain();
+            if (filterChain.contains(IDLE_FILTER)) {
+                filterChain.remove(IDLE_FILTER);
```

Comment:
I don't think gateway tracks in-use session idle for both http accept and connect sessions.